src/main/java/com/fidesmo/fdsm/Main.java

```diff
@@ -286,28 +326,34 @@ public static void main(String[] argv) {
                         }
                         fidesmoCard.deliverRecipe(client, formHandler, recipe);
                     } else if (args.has(OPT_UNINSTALL)) {
-                        CAPFile cap = CAPFile.fromStream(new FileInputStream((File) args.valueOf(OPT_UNINSTALL)));
-                        String recipe = RecipeGenerator.makeDeleteRecipe(cap.getPackageAID());
+                        String s = (String) args.valueOf(OPT_UNINSTALL);
+                        Path p = Paths.get(s);
+
+                        AID aid = null;
+                        if (!Files.exists(p)) {
+                            try {
+                                aid = AID.fromString(s);
+                            } catch (IllegalArgumentException e) {
+                                fail("Not a file or AID: " + s);
+                            }
+                        } else {
+                            aid = CAPFile.fromBytes(Files.readAllBytes(p)).getPackageAID();
```

Comment:
What if the file does not contain a valid AID? If an exception is thrown in such case, shouldn't we catch it?

Reply:
It will end with an IOException that gets caught and reported as:

`I/O error: Could not figure out the package name of the applet!`